src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -224,52 +224,53 @@ public static bool NotUsingHttps(string url)
-                    throw new ArgumentException("DatabaseName property has invalid value (null, empty or whitespace only)");
-                databaseRecord.DatabaseName = databaseRecord.DatabaseName.Trim();
-
-                if (databaseRecord.Settings.TryGetValue(RavenConfiguration.GetKey(x => x.Core.DataDirectory), out var dir))
+                
+                // Validate Directory
+                var dataDirectoryThatWillBeUsed = databaseRecord.Settings.TryGetValue(RavenConfiguration.GetKey(x => x.Core.DataDirectory), out var dir) == false ?
+                                         ServerStore.Configuration.Core.DataDirectory.FullPath : PathUtil.ToFullPath(dir, ServerStore.Configuration.Core.DataDirectory.FullPath);
+                
+                if (string.IsNullOrWhiteSpace(dir) == false)
                 {
-                    var requestedDirectory = PathUtil.ToFullPath(dir, ServerStore.Configuration.Core.DataDirectory.FullPath);
-
                     if (ServerStore.Configuration.Core.EnforceDataDirectoryPath)
                     {
-                        if (PathUtil.IsSubDirectory(requestedDirectory, ServerStore.Configuration.Core.DataDirectory.FullPath) == false)
+                        if (PathUtil.IsSubDirectory(dataDirectoryThatWillBeUsed, ServerStore.Configuration.Core.DataDirectory.FullPath) == false)
                         {
-                            throw new ArgumentException($"The administrator has restricted databases to be created only under the DataDir '{ServerStore.Configuration.Core.DataDirectory.FullPath}' but the actual requested path is '{requestedDirectory}'.");
+                            throw new ArgumentException($"The administrator has restricted databases to be created only under the DataDir '{ServerStore.Configuration.Core.DataDirectory.FullPath}'" +
+                                                        $" but the actual requested path is '{dataDirectoryThatWillBeUsed}'.");
                         }
                     }
 
-                    if (DataDirectoryInfo.CanAccessPath(requestedDirectory, out var error) == false)
+                    if (DataDirectoryInfo.CanAccessPath(dataDirectoryThatWillBeUsed, out var error) == false)
                     {
-                        throw new InvalidOperationException($"Cannot access path '{requestedDirectory}'. {error}");
+                        throw new InvalidOperationException($"Cannot access path '{dataDirectoryThatWillBeUsed}'. {error}");
                     }
                 }
+                
+                // Validate Name
+                databaseRecord.DatabaseName = databaseRecord.DatabaseName.Trim();
+                if (ResourceNameValidator.IsValidResourceName(databaseRecord.DatabaseName, dataDirectoryThatWillBeUsed, out string errorMessage) == false)
+                    throw new BadRequestException(errorMessage);
+               
+                if (LoggingSource.AuditLog.IsInfoEnabled)
```

Comment:
I would prefer to have this as soon as possible. Can we do it right away after deserialization?

Reply:
done - moved right after deserialization